### PR TITLE
Fix all lint issues instead of being prompted in several PR's about them

### PR DESCRIPTION
**Pull request by vipulnsward**
Fix all lint issues instead of being prompted in several PR's about them from codeclimate failures.

This fixes all failures except 2 from incompatible rules about case and indentation.

cc @rafaelfranca r? @fxn

### CODE REVIEW

**Comment by kaspth** [comment]
I'll close this because I've heard @fxn say he wants to manage these himself and just haven't had the time to get to them. Thanks!

**Comment by vipulnsward** [comment]
@kaspth this was actually based on the discussion with both of them

**Comment by kaspth** [comment]
Oh!

File: activemodel/lib/active_model/type/helpers/numeric.rb [hunk 1]
```diff
@@ -4,10 +4,14 @@ module Helpers
...
-                  when false then 0
-                  when ::String then value.presence
-                  else value
+                  when true then
+                    1
+                  when false then
+                    0
+                  when ::String then
+                    value.presence
+                    else
```
Comment (kaspth) [inline activemodel/lib/active_model/type/helpers/numeric.rb]: `else` is wrongly indented. I think we should strip the `then`s as well.

File: activemodel/lib/active_model/type/immutable_string.rb [hunk 2]
```diff
@@ -7,10 +7,14 @@ def type
...
-        when true then "t"
-        when false then "f"
-        else super
+        when ::Numeric, ActiveSupport::Duration then
+          value.to_s
+        when true then
+          "t"
+        when false then
+          "f"
+          else
```
Comment (kaspth) [inline activemodel/lib/active_model/type/immutable_string.rb]: Ditto.

File: activerecord/lib/active_record/associations/join_dependency.rb [hunk 3]
```diff
@@ -228,8 +228,8 @@ def walk(left, right)
         end
 
         def find_reflection(klass, name)
-          klass._reflect_on_association(name) or
-            raise ConfigurationError, "Can't join '#{ klass.name }' to association named '#{ name }'; perhaps you misspelled it?"
+          klass._reflect_on_association(name) ||
+            raise( ConfigurationError, "Can't join '#{ klass.name }' to association named '#{ name }'; perhaps you misspelled it?")
```
Comment (kaspth) [inline activerecord/lib/active_record/associations/join_dependency.rb]: Extra space before the open parenthesis. Shouldn't we also remove the spaces within the interpolations?

File: activerecord/lib/active_record/reflection.rb [hunk 4]
```diff
@@ -23,7 +23,7 @@ def self.create(macro, name, scope, options, ar)
                 HasOneReflection
               when :belongs_to
                 BelongsToReflection
-              else
+                else
```
Comment (kaspth) [inline activerecord/lib/active_record/reflection.rb]: I see this is the old Bad RuboCop routine 😁

File: activesupport/bin/generate_tables [hunk 5]
```diff
@@ -84,7 +84,7 @@ module ActiveSupport
 
         def create_composition_map
           @ucd.codepoints.each do |_, cp|
-            if !cp.nil? and cp.combining_class == 0 and cp.decomp_type.nil? and !cp.decomp_mapping.nil? and cp.decomp_mapping.length == 2 and @ucd.codepoints[cp.decomp_mapping[0]].combining_class == 0 and !@ucd.composition_exclusion.include?(cp.code)
+            if !cp.nil? && cp.combining_class == 0 && cp.decomp_type.nil? && !cp.decomp_mapping.nil? && cp.decomp_mapping.length == 2 && @ucd.codepoints[cp.decomp_mapping[0]].combining_class == 0 && !@ucd.composition_exclusion.include?(cp.code)
```
Comment (kaspth) [inline activesupport/bin/generate_tables]: Does RuboCop know how to safely replace the `and`s without messing up precedence? This example does look okay, it's just that there's many instances on this line.

File: activemodel/lib/active_model/type/immutable_string.rb (hunk 2 above)
Comment (vipulnsward) [inline activemodel/lib/active_model/type/immutable_string.rb]: Yeah, this is those "wrong" linting rules that causes this. Let me try to fix this.

File: activesupport/test/core_ext/duration_test.rb [hunk 6]
```diff
@@ -208,7 +208,10 @@ def test_to_json
   end
 
   def test_case_when
-    cased = case 1.day when 1.day then "ok" end
+    cased = case 1.day
+            when 1.day
+              "ok"
+            end
     assert_equal cased, "ok"
```
Comment (kaspth) [inline activesupport/test/core_ext/duration_test.rb]: Wrong assertion order by the way.

File: guides/rails_guides/markdown/renderer.rb [hunk 7]
```diff
@@ -72,7 +72,7 @@ def convert_notes(body)
                           "warning"
                         when "TIP"
                           "info"
-                        else
+                          else
```
Comment (kaspth) [inline guides/rails_guides/markdown/renderer.rb]: Stop that, copper!

File: railties/test/json_params_parsing_test.rb [hunk 8]
```diff
@@ -26,24 +26,24 @@ def self.name; 'Foo'; end
       end
     }
 
-    assert_nil app.call(make_env({ 't' => nil }))
-    assert_nil app.call(make_env({ 't' => [nil] }))
+    assert_nil app.call(make_env("t" => nil))
+    assert_nil app.call(make_env("t" => [nil]))
```
Comment (kaspth) [inline railties/test/json_params_parsing_test.rb]: Looks like this method call graduated high school and ditched the braces 🤓

**Comment by fxn** [comment]
I'll try to have a look at this one tonight.

**Comment by fxn** [comment]
There were a few details like some `(return)` and others that needed fine-tuning. I ran the auto corrections against current master and made a pass resulting in bb1ecdcc677bf6e68e0252505509c089619b5b90, which is essentially the same patch.

Vipul you were of course given credit. Thanks!

**Pull request closed without merging**